apollo-core/src/main/java/com/ctrip/framework/foundation/internals/DefaultProviderManager.java

```diff
@@ -16,35 +16,36 @@
  */
 package com.ctrip.framework.foundation.internals;
 
-import java.util.LinkedHashMap;
-import java.util.Map;
-
-import com.ctrip.framework.foundation.internals.provider.DefaultApplicationProvider;
-import com.ctrip.framework.foundation.internals.provider.DefaultNetworkProvider;
-import com.ctrip.framework.foundation.internals.provider.DefaultServerProvider;
+import com.ctrip.framework.apollo.core.spi.Ordered;
 import com.ctrip.framework.foundation.spi.ProviderManager;
+import com.ctrip.framework.foundation.spi.provider.ApplicationProvider;
+import com.ctrip.framework.foundation.spi.provider.NetworkProvider;
 import com.ctrip.framework.foundation.spi.provider.Provider;
+import com.ctrip.framework.foundation.spi.provider.ServerProvider;
+import java.util.LinkedHashMap;
+import java.util.Map;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class DefaultProviderManager implements ProviderManager {
+
   private static final Logger logger = LoggerFactory.getLogger(DefaultProviderManager.class);
   private Map<Class<? extends Provider>, Provider> m_providers = new LinkedHashMap<>();
 
   public DefaultProviderManager() {
     // Load per-application configuration, like app id, from classpath://META-INF/app.properties
-    Provider applicationProvider = new DefaultApplicationProvider();
+    Provider applicationProvider = ServiceBootstrap.loadPrimary(ApplicationProvider.class);
```

Comment:
As ProviderManager is loaded via ServiceBootstrap, I suppose it's not quite necessary to load ApplicationProvider with ServiceBootstrap here since user could customize the ProviderManager directly?

Reply:
Yes,   it is not necessary.  you can redefine`ProviderManager` to use your own`ApplicationProvider` implementation.   
 in this scene, I think `ApplicationProvider` can be loaded by ServiceBootstrap is the ability provided by  `DefaultProviderManager `.
 If I only want to redefine my own `ApplicationProvider` , I just need to write `ApplicationProvider` implementation
 by this ability. it is not conflict.
 Removing this feature, I also think it is ok.

Reply:
I think we could remove this feature for now. If we find it quite necessary in the future, we could add it back.